ecs.tf

```diff
@@ -78,6 +78,44 @@ module "app_container_definition" {
   }
 }
 
+module "datadog_container_definition" {
+  source                   = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name           = "datadog-agent"
+  container_image          = "datadog/agent:latest"
+  container_cpu            = "10"
+  container_memory         = "128"
+  essential                = false
```

Comment:
I take it that since this container isn't marked as essential it will just stop with an error code if the key isn't defined?

Reply:
This was leftover from the original implementation of this solution - I assume it's been marked false so the failure doesn't cascade if it can't start

Reply:
Unless the agent doesn't die and keeps hitting datadog without a token which might result in datadog blacklisting your ips?

Reply:
Another good point! I'm trying to find some info in the logs else I'll be able to pop on chat with datadog when they open up shop

Reply:
I've contacted datadog, here's the summary

When you have the API Key incorrectly added to or is a null result and is missing, the agent will try to connect to the platform. Retries are continuous and will continue for the life of the container. They should now affect performance as the agent overhead is very low. The IP will also not get blocked for sending continuous requests.

Reply:
With the ternary statement on the container_definitions we shouldn't add the datadog agent when we don't have an api key added

```container_definitions = var.datadog_api_key_from != "null" ? "[${module.app_container_definition.json_map},${module.datadog_container_definition.json_map}]" : "[${module.app_container_definition.json_map}]"```